src/Components/Server/src/ComponentHub.cs

```diff
@@ -202,7 +204,16 @@ private async void CircuitHost_UnhandledException(object sender, UnhandledExcept
 
             try
             {
-                Log.UnhandledExceptionInCircuit(_logger, circuitId, (Exception)e.ExceptionObject);
+                var exception = (Exception)e.ExceptionObject;
+                if (!IsBadInput(exception))
+                {
+                    Log.UnhandledExceptionInCircuit(_logger, circuitId, exception);
+                }
+                else
+                {
+                    Log.UnhandledExceptionFromBadInput(_logger, circuitId, exception);
+                }
```

Comment:
I really like that the intention is that we can tell the difference between user code throwing and framework code - I just want to make sure we're thinking about the problem holistically.

Reply:
Explanation in the comment above.